sycl/source/device_selector.cpp

```diff
@@ -31,6 +31,25 @@ device device_selector::select_device() const {
 }
 
 int default_selector::operator()(const device &dev) const {
+
+  // Take note of the SYCL_BE environment variable when doing default selection
+  const char *SYCL_BE = std::getenv("SYCL_BE");
+  std::string backend = (SYCL_BE ? SYCL_BE : "");
+  if (backend != "") {
```

Comment:
If would be more efficient (no string comparison) to check SYCL_BE for being NULL:

```suggestion
  if (SYCL_BE != nullptr) {
```